packages/@sanity/schema/src/descriptors/convert.ts

```diff
@@ -632,3 +633,32 @@ function maybeBoolean(val: unknown): boolean | undefined {
   }
   return undefined
 }
+
+function maybeI18n(val: unknown): LocalizedMessage | undefined {
+  if (!isObject(val) || Array.isArray(val)) return undefined
+
+  // Check if it's a valid I18nTextRecord format
+  const validI18n = Object.entries(val).every(([key, value]) => {
+    return (
+      typeof key === 'string' &&
+      isObject(value) &&
+      'key' in value &&
+      'ns' in value &&
+      typeof value.key === 'string' &&
+      typeof value.ns === 'string'
+    )
+  })
+
+  if (!validI18n) return undefined
+
+  // Convert I18nTextRecord to LocalizedMessage format
+  const localizedMessage: LocalizedMessage = {}
+  for (const [field, i18nValue] of Object.entries(val)) {
+    if (isObject(i18nValue) && 'key' in i18nValue && 'ns' in i18nValue) {
+      // Store the i18n reference as a string in format "ns:key"
+      localizedMessage[field] = `${i18nValue.ns}:${i18nValue.key}`
```

Comment:
@judofyr was this your intent with the `LocalizedMessage` type? Otherwise we can reuse the existing i18n object structure.

Reply:
Hmm... I'm not sure what I thought around LocalizedMessage. If `ns` and `key` are the fundamental fields then it's probably better to reuse, yes.